### DISCUSSION TITLE
MaryTTS support?

### DISCUSSION THREAD

**Discussion by zminer123** [erew123/alltalk_tts#152] [Q&A]
Hello there,
First of all, I'm sorry if my question is stupid. I am trying to set up MaryTTS in HomeAssistant, and one of the things that Coqui had added towards the end of its life was MaryTTS server support. I was wondering if the API you have is also compatible? Ideally, I'd like to be able to run your server, then link to it from my HomeAssistant setup to use it as a TTS.

I'm following this guide: https://www.youtube.com/watch?v=vtZbGrJqoZU, and hoping to essentially use your project instead of Coqui-TTS.

I appreciate any help or guidance!

**Comment by erew123** [comment]
Hi @zminer123 

AllTalk doesn't have any API compatibility with Mary TTS,s API. For reference I believe this is the [Mary API](https://marytts.github.io/javadoc/marytts/MaryInterface.html)

I cant imagine it would be too hard to write/code something for HomeAssistant, however, its not something I personally have space to do at the moment. I would imagine you should be able to manipulate/edit one of the other TTS integrations to work with a few changes. https://github.com/home-assistant/core/blob/dev/homeassistant/components/marytts/tts.py

Though saying that I have to admit I dont know the backend workings of HomeAssistant, so I dont know exactly how it makes its calls and handles replies/responses.

That aside, I may be adding an OpenAPI TTS compatible integration soon. So that may be an alternative as I see someone has written a HomeAssistant option https://github.com/sfortis/openai_tts

Thanks

**Comment by zminer123** [comment]
Thank you very much for the quick response. I eventually was able to use
the "tts-server" command from coqui to solve my issues after some minor
changes to the python to hard set some parameters I wanted. I still think
your solution is infinitely more elegant however, so if you do end up
adding open AI compatibility I will have to play with it and see if I can
get it working well.

Excellent work on the project in general by the way. I honestly can't
believe how quickly AI technology jumps. I started playing with it about a
year ago starting with stable diffusion, and the leaps and bounds it has
made have been utterly astounding.

Thanks again!
Zach
